pkg/collector/corechecks/servicediscovery/module/envs.go

```diff
@@ -153,3 +186,168 @@ func getEnvs(proc *process.Process) (map[string]string, error) {
 	}
 	return envs, nil
 }
+
+// EnvScanner reads the environment variables file in chunks.
+// Collects only those variables that match the target map if the map is not empty,
+// otherwise collect all environment variables.
+type EnvScanner struct {
+	file    *os.File          // open pointer to environment variables file
+	targets map[uint64]string // map of environment variables of interest
+	buffer  []byte            // buffer for reading a fragment from a file
+	end     int               // last index of data read
+	err     error             // last detected error
+	envs    map[string]string // collected environment variables
+	env     []byte            // the last processed environment variable
+}
+
+// NewEnvScanner returns a new [EnvScanner] to read from path.
+func NewEnvScanner(path string, bufSize int, targets []string) *EnvScanner {
```

Comment:
Why can't we just use `bufio.Scanner`? It's already used in other places in the agent for reading env variables and should be good enough in terms of performance and will result in much simpler code. The `NewTextScannerIterator` in `pkg/security/ptracer/proc.go` is exported so it could even be reused directly.

Reply:
I changed code to use bufio.Scanner